components/public-api-server/pkg/server/server.go

```diff
@@ -114,13 +131,26 @@ func Start(logger *logrus.Entry, version string, cfg *config.Configuration) erro
 
 	oidcService := oidc.NewService(cfg.SessionServiceAddress, dbConn, cipherSet, stateJWT)
 
+	var idpCache idp.KeyCache
+	if redisClient == nil {
+		log.Info("No Redis cache configured - caching IDP public keys in memory.")
```

Comment:
Do we actually want to allow this mode? Given we run multiple replicas of pub-api, this is bound to cause issues. I'd rather we say we have a dependency on Redis and we do not try to cache in memory